modules/logging/src/main/java/org/eclipse/xpanse/modules/logging/config/RestTemplateConfig.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.logging.config;
+
+import java.util.Collections;
+import org.eclipse.xpanse.modules.logging.RestTemplateLoggingInterceptor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Configuration class to create a RestTemplate bean.
+ */
+@Configuration
+public class RestTemplateConfig {
```

Comment:
It is better to have dedicated RestTemplate beans for each of our REST clients. This way, we can control it better. 
We need to have one in terraformboot, one for ZitadelIdentityProviderService and one for flexibleEngine.

in fact, the terraformboot is using its own version of RestTemplate - https://github.com/eclipse-xpanse/xpanse/blob/dae90cdf53fb8979a0f23643291a6cca84408702/modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/ApiClient.java#L644

We must update the interceptor directly  on the ApiClient bean. 

We must remove the RestTemplateConfig class and add the interceptors on each of the RestTemplates which we are creating.

Reply:
updated.